app/models/category.rb

```diff
@@ -0,0 +1,37 @@
+class Category < ActiveRecord::Base
+  belongs_to :user
+  has_many :articles
+  belongs_to :parent,
+    foreign_key: 'parent_id',
+    class_name: 'Category'
+
+  has_many :subcategories,
+    foreign_key: 'parent_id',
+    class_name: 'Category'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.